validation/linux_cgroups_blkio.go

```diff
@@ -1,30 +1,100 @@
 package main
 
 import (
+	"fmt"
+	"runtime"
+
+	"github.com/mndrix/tap-go"
 	"github.com/opencontainers/runtime-tools/cgroups"
 	"github.com/opencontainers/runtime-tools/validation/util"
 )
 
-func main() {
+func testBlkioCgroups() error {
+	var weight uint16 = 500
+	var leafWeight uint16 = 300
+
+	t := tap.New()
+	t.Header(0)
+	defer t.AutoPlan()
+
+	// It's assumed that a device /dev/sda (8:0) exists on the test system.
+	// The minor number must be always 0, as it's only allowed to set blkio
+	// weights to a whole block device /dev/sda, not to partitions like sda1.
+	cases := []struct {
+		major int64
+		minor int64
+		rate  uint64
+	}{
+		{8, 0, 102400},
+		{8, 0, 204800},
+	}
+
+	for _, c := range cases {
+		g, err := util.GetDefaultGenerator()
+		if err != nil {
+			return err
+		}
+
+		g.SetLinuxCgroupsPath(cgroups.AbsCgroupPath)
+		g.SetLinuxResourcesBlockIOWeight(weight)
+		g.SetLinuxResourcesBlockIOLeafWeight(leafWeight)
+
+		g.AddLinuxResourcesBlockIOWeightDevice(c.major, c.minor, weight)
+		g.AddLinuxResourcesBlockIOLeafWeightDevice(c.major, c.minor, leafWeight)
+		g.AddLinuxResourcesBlockIOThrottleReadBpsDevice(c.major, c.minor, c.rate)
+		g.AddLinuxResourcesBlockIOThrottleWriteBpsDevice(c.major, c.minor, c.rate)
+		g.AddLinuxResourcesBlockIOThrottleReadIOPSDevice(c.major, c.minor, c.rate)
+		g.AddLinuxResourcesBlockIOThrottleWriteIOPSDevice(c.major, c.minor, c.rate)
+
+		err = util.RuntimeOutsideValidate(g, t, util.ValidateLinuxResourcesBlockIO)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func testEmptyBlkio() error {
```

Comment:
I think it could be part of testBlkioCgroups. We can add more args to the ```cases struct```.  Take https://github.com/opencontainers/runtime-tools/blob/master/validation/state.go#L29 for example.

Reply:
@liangchenye That sounds like a good idea. Will try to do that.